### ISSUE TITLE
Wrong documentation when there are multiple functions with the same name

### ISSUE THREAD

**Issue by segevfiner** [microsoft/vscode-go#2107]
While Go doesn't support function overloading, you can still have multiple functions with the same name but with a *different* receiver. For example: `testing` has the `Run` function.

It seems that `gocode` and `godef` don't supply us with the receiver. But `go doc` does support receiving it, like so `go doc testing T.Run`.

Steps to Reproduce:

1. Save as `foo_test.go`:
```go
package footest

import "testing"

func TestBla(t *testing.T) {
	t.Ru
}
```
2. Try to complete `t.Ru`.
3. You will get the documentation for `testing.B.Run` instead of `testing.T.Run`.

Run: `gocode -f=json -builtin -in=foo_test.go autocomplete 74 | python -m json.tool` to see what `gocode` gives us, `go doc testing Run` to see what we get from `go doc`, and `go doc testing T.Run` for what we would have liked to get.

vscode-go v0.7.0

**Comment by ramya-rao-a** [comment]
@segevfiner Thanks for reporting!

This is a hard problem to solve...

We cant use `go doc testing T.Run` because there is no way to know that we should use `T.Run` and not `Run`. We dont get that information from `gocode`

`gocode` does give the function signature which I thought I could use to get the right info from `go doc testing Run` which gives docs for all 3 Run() implementations.

But this wouldnt work as `gocode` gives us the signature as `func(name string, f func(t *testing.T)) bool` where as `go doc testing Run` returns `func (t *T) Run(name string, f func(t *T)) bool`

i.e since T is qualified with the package name in the gocode output but not the one from go doc, its hard to get a match.

> You will get the documentation for testing.B.Run instead of testing.T.Run.

Actually the documentation for all the 3 Run() are shown (without the signature)

cc @segevfiner

**Comment by segevfiner** [comment]
As I said, we will need to extend `gocode` and `godef` so that the also supply us with the receiver. Keeping backwards compatibility in mind, so we might have to add a flag so that they only print the receiver when asked.

**Comment by ramya-rao-a** [comment]
Based on the current state, the problem for when using `gocode` was fixed with #2215

This leaves the docs that appear on hover.

@segevfiner Does this problem occur when using the language server as well?
If not, then I am inclined to close this issue in favor of using the language server

cc @stamblerre

**Comment by stamblerre** [comment]
The language server does not have this problem, so I think we can close this issue.
